addons/mail/models/mail_activity.py

```diff
@@ -735,6 +736,67 @@ def _compute_activity_state(self):
             else:
                 record.activity_state = False
 
+    def _search_activity_state(self, operator, value):
+        all_states = {'overdue', 'today', 'planned', False}
+        if operator == '=':
+            states = {value}
```

Comment:
Maybe states could be rename to search_states, so that we have
  * all_states: all possible value
  * search_states: values we search
  * search_states_int: integer version of values we search